server_src/handlers/generic-websocket.adapter.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * An OctoFarm websocket handler suitable for any websocket application like OctoPrint, Mariner, Klipper, etc.
+ * Copy it to fit your needs, or take a look at 'octoprint-websocket.adapter.js'
+ */
+module.exports = class GenericWebsocketAdapter {
```

Comment:
Nice!

Reply:
Still very clunky, just to give an idea. Also the adapter design is currently quite bad, will need to work on that

Reply:
Excuse my ignorance but why bad?

Reply:
Bad in the awilix context, so bit hard to explain as I cant put my finger on it. I guess I need the P.O.V. from a potential OF plugin dev to truly understand if this adapter is going to help them.

Reply:
> Bad in the awilix context, so bit hard to explain as I cant put my finger on it. I guess I need the P.O.V. from a potential OF plugin dev to truly understand if this adapter is going to help them.

For me, if it helps, when I'm writing the connectors for the others I want to be able to open up a websocket, pass the listener functions and have the resulting information go through the cleaners. Now an issue I've always thought of is making this information available to the rest of the system, and also the UI changing slightly for each type of web interface. 

I'd gather we'd have a preset set of variables anyone using the adapters must stick to to be in keeping with the current backend. New variables could be added / removed based on the type of printer.  

Resin printers won't need temp, and half the control options for example. 
Klipper the file stuff would be slightly different (I think)